spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonDeserializer.java

```diff
@@ -206,9 +215,11 @@ public void configure(Map<String, ?> configs, boolean isKey) {
 			if (isKey && configs.containsKey(KEY_DEFAULT_TYPE)) {
 				if (configs.get(KEY_DEFAULT_TYPE) instanceof Class) {
 					this.targetType = (Class<T>) configs.get(KEY_DEFAULT_TYPE);
+					this.reader = this.objectMapper.readerFor(this.targetType);
```

Comment:
Hm. Looks like we don't need `this.reader = null;` in the ctor at all! 😃

Reply:
Yeah - I had it final but tests failed because of this mode.